packages/access-client/src/agent-data.js

```diff
@@ -111,12 +111,15 @@ export class AgentData {
 
   /**
    * @deprecated
-   * @param {import('@ucanto/interface').DID} did
+   * @param {import('@ucanto/interface').DID<'key'>} did
    * @param {import('./types').SpaceMeta} meta
    * @param {import('@ucanto/interface').Delegation} [proof]
    */
   async addSpace(did, meta, proof) {
     this.spaces.set(did, meta)
+    if (!this.currentSpace) {
```

Comment:
It's not obvious to me that calling `addSpace` would also have the side effect of changing `this.currentSpace`, either from the comments or tests.

Why should this have that side effect, instead of leaving this to the caller to decide whether or not to call `Agent#setCurrentSpace`? The latter seems like the most flexible solution.

Reply:
(and, even if it is desirable to introduce the side effect here, why would this do it via assigning `this.currentSpace =` vs calling `this.setCurrentSpace`?)